### PR TITLE
ci: Add PR linting

### PR DESCRIPTION
**Pull request by petermakowski**
## Done
This PR adds a PR Lint action that enforces conventional commits convention for pull request titles. 

- ci: add pr lint

### Storybook

To see rendered examples of all react-components, run:

```shell
yarn start
```

### QA in your project

from `react-components` run:

```shell
yarn build
npm pack
```

Install the resulting tarball in your project with:

```shell
yarn add <path-to-tarball>
```

### QA steps

- Steps for QA.

## Fixes

Fixes: # .

### CODE REVIEW

**Comment by bartaz** [comment]
@petermakowski Would it be worth documenting something what is a valid PR name? or is that clear enough when it fails?

File: .github/workflows/pr-lint.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  pull_request:
+    types: [opened, edited, synchronize, reopened]
+
+permissions:
+  pull-requests: read
+
+jobs:
+  main:
+    name: Validate PR title
+    runs-on: ubuntu-22.04
```
Comment (bartaz) [inline .github/workflows/pr-lint.yml:13]: any reason not to use latest?

**Review by bartaz — APPROVED** [review]
Couple small questions in comments to discuss if needed, but it's fine as is as well.

**Comment by petermakowski** [comment]
> Couple small questions in comments to discuss if needed, but it's fine as is as well.

Thanks, will think about updating the docs with more info next.

**Comment by bartaz** [comment]
> > Couple small questions in comments to discuss if needed, but it's fine as is as well.
> 
> Thanks, will think about updating the docs with more info next.

Thanks. I tested how it fails, and error message is quite informative, so it's not a high priority, but once we get to the point of having automatic releases set up it would be good to have the complete process described somewhere.

File: .github/workflows/pr-lint.yml (hunk 1 above)
Comment (petermakowski) [reply .github/workflows/pr-lint.yml:13]: We generally try to use exact dependencies for predictable runs. Happy to change if you feel strongly about this.

**Pull request merged**